encoder/src/main/java/openlr/encoder/routesearch/SecondShortestRouteChecker.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    /**
+     *
+     * @param lengthAlongSecondShortestRoute length of the second shortest route
+     * @param lengthAlongLocation length of the route along location to which the second shortest route gives alternative to
+     * @return true: if the length of the second shortest route is not greater than route along location by relative threshold percentage
+     *         false: if the length of the second shortest route is greater than route along location by relative threshold percentage
+     */
+    boolean VerifyThreshold(int lengthAlongSecondShortestRoute, int lengthAlongLocation){
+        return (lengthAlongSecondShortestRoute < (lengthAlongLocation + (int) (lengthAlongLocation * relativeThreshold)));
+    }
+
+
+    /**
+     * @param length heuristic length
+     * @return true  if the heuristic is less than the maximum  length
+     *         false if the heuristic is greater than the maximum  length
+     */
+    private boolean networkLengthFilter(Integer length) {
+        return (length < maxLengthAllowed);
+    }
+
+    /**
+     * @param lastLineIndex index of the road segment in the route
+     * @return length along the location from start till the road segment at the lastLineIndex
+     */
+    private int lengthAlongLocation(int lastLineIndex) {
+        int routeLength = 0;
+        for (int index = 0; index <= lastLineIndex; ++index) {
+            routeLength += location.get(index).getLineLength();
+        }
+        return routeLength;
+    }
+
+    /**
+     *
+     * @param index index of the road segment in the location between the two location reference points
+     * @return true if an alternate path of length under the the relative threshold exist
+     *         false if no alternate path of length under the the relative threshold exist
+     */
+    public boolean hasValidDeviationBefore(int index) {
+        if (index > 0 && index < location.size() - 1) {
+            Set<Long> closedSet = new HashSet<>();
+            closedSet.add(location.get(index).getID());
+            RouteSearchData data = new RouteSearchData();
+            Line parentLine = location.get(index - 1);
+            int lengthCoveredByParentLine = lengthAlongLocation(index - 1);
+            int heuristics = lengthCoveredByParentLine + (int) GeometryUtils.distance(parentLine.getEndNode().getGeoCoordinates(),
+                    destination);
+            PQElem parent = new PQElem(parentLine, heuristics, lengthCoveredByParentLine, null);
+            data.addToOpen(parent);
+            return exploreNetworkForSecondShortestRouteUnderThreshold(closedSet, data, index);
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
+     *
+     * @param current parent road segment
+     * @param closedSet Set of road segments to skip
+     * @return list of successor road segments
+     */
+    private List<Line> getAcceptableSuccessors(PQElem current, Set<Long> closedSet) {
```

Comment:
I would rename the method to: ** calculateAcceptableSuccessors** 
Rename **current** to **currentRoadElement**
Rename ** closedSet** to ** closedRoadSegmentSet**

Reply:
ac5407dc43f6f89b768b92e9571ed9f04942a5d2